vhpidirect/quickstart/intvector/caux.c

```diff
@@ -0,0 +1,16 @@
+#include <stdio.h>
+#define SIZE_ARRAY (sizeof(intArray)/sizeof(int))
+
+int intArray[6];
+
+int getIntArrSize(){//function acts like a constructor so initialise the variable
+    return SIZE_ARRAY;
+}
+
+int* getIntArr_ptr(){//function acts like a constructor so initialise the variable
+    for (int i = 0; i < SIZE_ARRAY; i++)
```

Comment:
So, I did not really understand what you were expecting from the main.c+caux subexample, but I think you may like what I posted. I have not run through the many ways to compile and run the executable, as it is an early example, and we have a dedicated section for that, and the executable asks for user input.

The subexample prompts the user for the array's length, fills the array, and calls ghdl_main. Only thing I don't like is trusting the user to enter a reasonable number

Reply:
> I have not run through the many ways to compile and run the executable, as it is an early example, and we have a dedicated section for that, and the executable asks for user input.

My point was not about different ways to compile and run the executable, but about different requirements dependending on the number of C files.

- `-Wl,main.c` or `-Wl,caux.c` only. Already covered.
- `-Wl,caux.c -Wl,main.c`. Not used yet. Questions that users might have:
  - Is it `-I./` required?
  - Does `caux.h` need to exist?

Of course, this can be explained in a different example instead of here.

> The subexample prompts the user for the array's length, fills the array, and calls ghdl_main.

I think it would be better to use/parse `argv[1]`. Three options:

- If `argc == 1`, use `argv[1]` as an integer (`atoi`).
- Pase `argv` and look for `--size=` or `-s`. Then use `atoi` as in the previous point.
- Search for `--` in argv and consider all previous args to be "custom args for C". Then search for `--size` or `-s` as in the previous point. Pass args after `--` to GHDL. Open question would be: what to do when `--` is not found? Are all args for C or for ghdl_main?

Either of these would be interesting because they'd show how to add custom args (that need to be removed from the argv passed to GHDL). I would not use any lib such as optargs, tho. We want to support a single custom argument.

> Only thing I don't like is trusting the user to enter a reasonable number, but I cannot seem to break the executable with outrageous values so I think it is safe.

Even if they try to, it's ok. See ghdl/ghdl#822.

> Only thing is getting the test to provide runtime stdin input...

I think it is better to pass argv, which allows `./tb 26` or `./tb -s 26` or `./tb --size=26` or `./tb --size 26 -- --wave=mywave.ghw`.